sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/theme/Theme.kt

```diff
@@ -1,17 +1,66 @@
 package co.nimblehq.sample.compose.ui.theme
 
-import androidx.compose.material.MaterialTheme
-import androidx.compose.material.lightColors
-import androidx.compose.runtime.Composable
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.material.*
+import androidx.compose.runtime.*
 
 @Composable
 fun ComposeTheme(
+    darkTheme: Boolean = isSystemInDarkTheme(),
     content: @Composable () -> Unit
 ) {
-    MaterialTheme(
-        colors = lightColors(
-            primary = GreenCitrus
-        ),
-        content = content
-    )
+    val colors = if (darkTheme) {
+        DarkColorPalette
+    } else {
+        LightColorPalette
+    }
+    val dimensions = LocalAppDimensions.current
+    val styles = LocalAppStyles.current
+    val typography = LocalAppTypography.current
+    val shapes = Shapes()
+
+    CompositionLocalProvider(
+        LocalColors provides colors,
+        LocalAppDimensions provides dimensions,
+        LocalAppStyles provides styles
+    ) {
+        MaterialTheme(
+            colors = colors.themeColors,
+            typography = typography,
+            shapes = shapes,
+            content = content
+        )
+    }
+}
+
+/**
+ * Alternate to [MaterialTheme] allowing us to add our own theme systems
+ * or to extend [MaterialTheme]'s types e.g. return our own [Colors] extension.
+ */
+object AppTheme {
+
+    val colors: AppColors
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalColors.current
+
+    val typography: Typography
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalAppTypography.current
+
+    val shapes: Shapes
+        @Composable
+        @ReadOnlyComposable
+        get() = MaterialTheme.shapes
+
+    val dimensions: AppDimensions
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalAppDimensions.current
+
+    val styles: AppStyles
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalAppStyles.current
```

Comment:
Hmmmm, what do you mean by "Alternate"? 🙏

Reply:
@Tuubz That means we will access all theme resources via `AppTheme` class instead of the original `Theme` class. The idea is the replace the Theme by AppTheme.